core/src/main/java/hudson/EnvVars.java

```diff
@@ -427,4 +506,33 @@ private static EnvVars initMaster() {
             vars.remove("MAVEN_OPTS");
         return vars;
     }
+
+    /**
+     * Return environment variables for launching a native process.
+     * 
+     * In Unix, does nothing.
+     * 
+     * In Windows, removes environment variables with case-different names
+     * except only one value.
+     * It is implementation-dependent which value is used.
+     * 
+     * @return environment variables adjusted for a native process.
+     */
+    public EnvVars forNativeProcess() {
+        if (platform.envCaseSensitive || !isCaseSensitive()) {
```

Comment:
Windows slaves when run through cygwin sshd will have strange behavior.
They are not UNIX and will be case insensitive, but running shell scripts on them, they will behave differently to the same shell script on Unix - which will have case sensitive values - which will just lead to confusion and bug reports.